include/nbl/system/ISystem.h

```diff
@@ -266,33 +274,36 @@ class ISystem final : public core::IReferenceCounted
     }
 
     //! Warning: blocking call
-    core::smart_refctd_ptr<IFileArchive> createFileArchive(const std::filesystem::path& filename)
+    core::smart_refctd_ptr<IFileArchive> createFileArchive(const std::filesystem::path& filename, const std::string_view& password = "")
```

Comment:
lets call it `openFileArchive` for now, as we dont support writing to archives (unlike files) for now.